src/__test__/Media.test.tsx

```diff
@@ -530,6 +530,46 @@ describe("Media", () => {
     })
   })
 
+  describe("prevent nested unnecessary renders", () => {
+    it("only renders one element when Media is nested within Media", () => {
+      const query = renderer.create(
+        <MediaContextProvider>
+          <Media at="extra-small">
+            <span className="extra-small" />
+          </Media>
+
+          <Media at="medium">
+            <Media at="extra-small">
+              <span className="extra-small" />
+            </Media>
+            <Media at="medium">
+              <span className="medium" />
+            </Media>
+            <Media at="large">
+              <span className="large" />
+            </Media>
+          </Media>
+
+          <Media at="large">
+            <span className="large" />
+          </Media>
+        </MediaContextProvider>
+      )
+
+      expect(
+        query.root.findAllByProps({ className: "extra-small" }, { deep: true })
+          .length
+      ).toBe(1)
```

Comment:
I would feel slightly more sure about this test if you were to omit the non-nested `extra-small` component and just test that there are `0` occurrences from the nested components.

Reply:
Makes sense!

Reply:
👉 https://github.com/artsy/fresnel/pull/158/commits/f0384a8da4cf2f8f4581a923ef4dd4062b5a4266